### DISCUSSION TITLE
Cannot connect UART0 to PMS5003 sensor

### DISCUSSION THREAD

**Discussion by kengineer23** [micropython/micropython#14489] [ESP32]
I am working on a product which involves getting the AQI data and i was trying to connect ESP32's UART0 channel(Tx=1. Rx=3) with PMS5003, however whenever I try to run the code on Thonny IDE with the above config, it shows the error "Could not interrupt current process. Please wait, try again or select Stop/Restart!". However it runs fine with UART2 channel. The problem is I cannot use UART2 channel as I need to develop the code according to the PCB that has already been developed by the team and have to use UART0 only. Now one issue, I think is that when connected to the computer with a USB, it uses the UART0 channel for uploading the program hence the UART0 cannot be used for other purpose. Please can anyone help me with this issue. I'll attach the code:

`pms5003 = PMS5003(uart=UART(0, tx=Pin(1), rx= Pin(3), baudrate=9600),pin_reset=Pin(4), pin_enable=Pin(5), mode="active")`

this is the UART declaration

**Comment by peterhinch** [comment]
I think you may have to modify the PCB with a scalpel and soldering iron :(
